streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -570,6 +581,21 @@ void runLoop() {
         }
     }
 
+    public void initiateShutdown(){
+        shutdownRequested.set(true);
```

Comment:
can not directly call sendShutdownRequest because the consumer should only accessed from the stream thread